core/src/cluster_info.rs

```diff
@@ -841,12 +841,13 @@ impl ClusterInfo {
                     .lookup(&CrdsValueLabel::ContactInfo(self.id()))
                     .unwrap_or_else(|| panic!("self_id invalid {}", self.id()));
 
-                pulls.push((
-                    entrypoint.id,
-                    self.gossip.pull.build_crds_filter(&self.gossip.crds),
-                    entrypoint.gossip,
-                    self_info.clone(),
-                ))
+                self.gossip
+                    .pull
+                    .build_crds_filters(&self.gossip.crds, CRDS_GOSSIP_PULL_SPLIT_COUNT)
```

Comment:
Do we want to send all CRDS_GOSSIP_PULL_SPLIT_COUNT queries at the same time? This could be a large number if the number of items grows large (assuming we don't cap CRDS_GOSSIP_PULL_SPLIT_COUNT at 4)